### ISSUE TITLE
[Accessibility]: Search result repo and path/commit message completely cut off on narrow screens

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#36322]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

Search result repo and path/commit message completely cut off on narrow screens. This happens for all result types. 

![image](https://user-images.githubusercontent.com/206864/166076583-feea0aa1-5401-45a3-acc0-a519412bf9e7.png)

### Expected behavior

Design team to suggest a new design for narrow screens (wrapping? horizontal scrolling?)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by varungandhi-src** [comment]
Related issue with some overlap:
- https://github.com/sourcegraph/sourcegraph/issues/36629
